### ISSUE TITLE
GeneratorFunction should not be newable

### ISSUE THREAD

**Issue by LongTengDao** [microsoft/TypeScript#56645]
### ⚙ Compilation target

ES2015 ES2018

### ⚙ Library

generator asyncgenerator

### Missing / Incorrect Definition

`GeneratorFunction` and `AsyncGeneratorFunction` (`async function * () {}`) should not be newable.

`GeneratorFunction` means `function * () {}` which is only callable, while `GeneratorFunctionConstructor` means `function * () {}.constructor` which is both callable and newable.

### Sample Code

```TypeScript
declare const gen: GeneratorFunction;
new gen;// there should cause a type error, but not
```
